misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -76,60 +80,94 @@ public CoverageReport(Path sourceJarPath, Path targetDirPath, String orgName, St
                 for (ISourceFileCoverage sourceFileCoverage : packageCoverage.getSourceFiles()) {
-                    if (sourceFileCoverage.getName().contains(BLangConstants.BLANG_SRC_FILE_SUFFIX) &&
-                            !sourceFileCoverage.getName().contains("tests/")) {
+
+                    // Extract the Module name individually for each source file
+                    // This is done since some source files come from other modules
+                    // sourceFileCoverage : "<orgname>/<moduleName>:<version>
+                    String sourceFileModule = sourceFileCoverage.getPackageName().split("/")[1];
+
+                    if (sourceFileCoverage.getName().contains(BLangConstants.BLANG_SRC_FILE_SUFFIX)
+                            && !sourceFileCoverage.getName().contains("tests/")) {
                         List<Integer> coveredLines = new ArrayList<>();
                         List<Integer> missedLines = new ArrayList<>();
+
                         for (int i = sourceFileCoverage.getFirstLine(); i <= sourceFileCoverage.getLastLine(); i++) {
                             ILine line = sourceFileCoverage.getLine(i);
-                            if (line.getInstructionCounter().getTotalCount() == 0
-                                    && line.getBranchCounter().getTotalCount() == 0) {
+
+                            if (line.getInstructionCounter().getTotalCount() == 0 &&
+                                    line.getBranchCounter().getTotalCount() == 0) {
                                 // do nothing. This is to capture the empty lines
-                            } else if ((line.getBranchCounter().getCoveredCount() == 0
-                                    && line.getBranchCounter().getMissedCount() > 0)
-                                    || line.getStatus() == NOT_COVERED) {
+                            } else if ((line.getBranchCounter().getCoveredCount() == 0 &&
+                                    line.getBranchCounter().getMissedCount() > 0) || line.getStatus() == NOT_COVERED) {
                                 missedLines.add(i);
                             } else if (line.getStatus() == PARTLY_COVERED || line.getStatus() == FULLY_COVERED) {
                                 coveredLines.add(i);
                             }
                         }
-                        ModuleCoverage.getInstance().addSourceFileCoverage(moduleName, sourceFileCoverage.getName(),
-                                coveredLines, missedLines);
+
+                        // Only add the source files that belong to the same module
+                        if (sourceFileModule.equals(moduleName)) {
+                            ModuleCoverage.getInstance().addSourceFileCoverage(sourceFileModule,
+                                    sourceFileCoverage.getName(), coveredLines, missedLines);
+                        } else {
+                            String jsonCachePath = this.jsonCache.toString() + "/"
```

Comment:
The module name can include "." as well right? 
Let's see we have a project with 2 modules as "module_test" & "module.test", will this have an impact?

Reply:
Yes. This certainly is a problem. Although it doesn't cause any impact to the overall functionality, it still creates unnecessary cache files when a module has '_' in its name. 

I will fix this to only consider the version when replacing '_' with '.'

Reply:
I have added a function that deals with extracting the version and replacing the `_` with `.` 
This should only apply to the version and the module name will be spared.